File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/indexing/TransformerDefinition.java
package net.ravendb.abstractions.indexing;

public class TransformerDefinition implements Cloneable {
  private String transformResults;
  private int indexId;
  private String name;


  public int getIndexId() {
    return indexId;
  }



  public void setIndexId(int indexId) {
    this.indexId = indexId;
  }


  @Override
  public String toString() {
    if (name != null) {
      return name;
    }
    return transformResults;
  }


  @Override
  public Object clone() throws CloneNotSupportedException {
    TransformerDefinition clone = new TransformerDefinition();
    clone.setTransformResults(transformResults);
    clone.setName(name);
    return clone;
  }


  @Override
  public int hashCode() {
    final int prime = 31;
    int result = 1;
    result = prime * result + ((transformResults == null) ? 0 : transformResults.hashCode());
    return result;
  }
  @Override
  public boolean equals(Object obj) {
    if (this == obj)
      return true;
    if (obj == null)
      return false;
    if (getClass() != obj.getClass())
      return false;
    TransformerDefinition other = (TransformerDefinition) obj;
    if (transformResults == null) {
      if (other.transformResults != null)
        return false;
    } else if (!transformResults.equals(other.transformResults))
      return false;
    return true;
  }
  public String getTransformResults() {
    return transformResults;
  }
  public void setTransformResults(String transformResults) {
    this.transformResults = transformResults;
  }
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }


}

File: Raven.Abstractions/Indexing/TransformerDefinition.cs
namespace Raven.Abstractions.Indexing
{
    public class TransformerDefinition
    {
        /// <summary>
        /// Projection function.
        /// </summary>
        public string TransformResults { get; set; }

        /// <summary>
        /// Transformer identifier (internal).
        /// </summary>
        public int TransfomerId { get; set; }

        /// <summary>
        /// Transformer name.
        /// </summary>
        public string Name { get; set; }

        public TransformerLockMode LockMode { get; set; }

        /// <summary>
        /// Transformer version, used in transformer replication in order to identify if two transformers are indeed the same.
        /// </summary>
        public int? TransformerVersion { get; set; }

        public bool Equals(TransformerDefinition other)
        {
            return string.Equals(TransformResults, other.TransformResults);
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != GetType()) return false;
            return Equals((TransformerDefinition) obj);
        }

        public override int GetHashCode()
        {
            return (TransformResults != null ? TransformResults.GetHashCode() : 0);
        }

        public TransformerDefinition Clone()
        {
            return (TransformerDefinition) MemberwiseClone();
        }

        public override string ToString()
        {
            return TransformResults;
        }
    }
}
